addons/default/visiosoft/advs-module/src/Http/Controller/OptionConfigurationController.php

```diff
@@ -31,6 +32,30 @@ public function create(OptionConfigurationFormBuilder $form)
         return $form->render();
     }
 
+    public function ajaxCreate(OptionConfigurationRepositoryInterface $optionConfigurationRepository){
+        $parameters = $this->request->all();
+        $option_json = array();
+        foreach ($parameters as $key => $parameter_value) {
+            if ((strpos($key, "option-") === 0)) {
+                if ($parameter_value !== '') {
+                    $option_id = substr($key, 7);
+                    $option_json[$option_id] = $parameter_value;
+                }
+                unset($parameters[$key]);
+            }
+        }
+        $option_json = ['option_json' => json_encode($option_json)];
+        $configration = array_merge($parameters, $option_json);
+        $entry = $optionConfigurationRepository->create($configration);
+        $entry['currency_price'] = app(Currency::class)->format($entry->price, $entry->currency);
+
+        return $this->response->json($entry);
+    }
+
+    public function ajaxDelete(OptionConfigurationRepositoryInterface $optionConfigurationRepository){
+        return $optionConfigurationRepository->deleteConfig($this->request->id);
+    }
```

Comment:
Anyone can delete anyone's config! You need to add an auth middleware and check if the ad belongs to the authenticated user